### DISCUSSION TITLE
How can data be sent to onDone of a compound hierrachial state

### DISCUSSION THREAD

**Discussion by dsathyakumar** [statelyai/xstate#2098] [Q&A] [answered]
Hi ,
I have an `onDone` added to a a compound state (which has sub-states within it), to trigger it when a final state is reached among the sub-states. But, currently, the data prop of the event received is undefined. Is ter any way data can be passed to this ?

**Comment by Andarist** [comment, marked as answer]
You can attach data to the final state and it will become part of the done event. Examples here:
https://github.com/davidkpiano/xstate/blob/14cdef612df29b9e067a34ca96b0750e729c9aff/packages/core/test/final.test.ts#L25-L28
https://github.com/davidkpiano/xstate/blob/14cdef612df29b9e067a34ca96b0750e729c9aff/packages/core/test/invoke.test.ts#L41

**Reply by dsathyakumar** [reply]
Thanks @Andarist  Let me try this out. 👍
